llvm/lib/Target/RISCV/RISCVInstrInfoXCheri.td

```diff
@@ -295,7 +297,7 @@ def CToPtr      : Cheri_rr<0x12, "ctoptr", GPR, GPCRC0IsDDC>;
 def CFromPtr    : Cheri_rr<0x13, "cfromptr", GPCR, GPR, GPCRC0IsDDC>;
 }
 def CSub        : Cheri_rr<0x14, "csub", GPR, GPCR>;
-let defsCanBeSealed = 1 in
+let defsCanBeSealed = 1, isAsCheapAsAMove = 1, isMoveReg = 1, isReMaterializable = 1 in
```

Comment:
```suggestion
let isMoveReg = 1, isReMaterializable = 1, isAsCheapAsAMove = 1,
    defsCanBeSealed = 1 in
```

Reply:
I listed them alphabetically, what's better about the other order?

Reply:
The vast majority of instances in the tree have `isReMaterializable` before `isAsCheapAsAMove`, including all the RISC-V ones (and the ones you added above...). Don't really care whether `isMoveReg` comes before or after that pair, just thought it looked nicer having it first, and I put `defsCanBeSealed` on the line below since it'll go away eventually and it looks better having the short line at the end rather than the start.

Reply:
Also the RISC-V backend doesn't currently make use of `isMoveReg` so technically it's not doing anything at the moment, but right to add it.

Reply:
Consistency is good :)